plugins/media-keys/msd-media-keys-manager.c

```diff
@@ -322,70 +331,70 @@ update_kbd_cb (GSettings           *settings,
-		g_free (signal_name);
-
-		if (keys[i].settings_key != NULL) {
-			tmp = g_settings_get_string (manager->priv->settings, keys[i].settings_key);
-		} else {
-			tmp = g_strdup (keys[i].hard_coded);
-		}
-
-		if (!is_valid_shortcut(tmp))
-		{
-			g_debug("Not a valid shortcut: '%s'", tmp);
-			g_free(tmp);
-			continue;
-		}
-
-		key = g_new0(Key, 1);
-
-		if (!egg_accelerator_parse_virtual(tmp, &key->keysym, &key->keycodes, &key->state))
-		{
-			g_debug("Unable to parse: '%s'", tmp);
-			g_free(tmp);
-			g_free(key);
-			continue;
-		}
-
-		g_free(tmp);
-
-		keys[i].key = key;
-
-		need_flush = TRUE;
-		grab_key_unsafe(key, TRUE, manager->priv->screens);
-	}
-
-	if (need_flush)
-	{
-		gdk_display_flush (dpy);
-	}
-
-	if (gdk_x11_display_error_trap_pop (dpy))
-	{
-		g_warning("Grab failed for some keys, another application may already have access the them.");
-	}
-
-	mate_settings_profile_end(NULL);
+        int i;
+        GdkDisplay *dpy;
+        gboolean need_flush = FALSE;
+
+        mate_settings_profile_start(NULL);
+
+        dpy = gdk_display_get_default ();
+        gdk_x11_display_error_trap_push (dpy);
+
+        for (i = 0; i < HANDLED_KEYS; i++)
+        {
+                char* tmp;
+                Key* key;
+
+                gchar* signal_name;
+                signal_name = g_strdup_printf ("changed::%s", keys[i].settings_key);
+                g_signal_connect (manager->priv->settings,
+                                  signal_name,
+                                  G_CALLBACK (update_kbd_cb),
+                                  manager);
+                g_free (signal_name);
+
+                if (keys[i].settings_key != NULL) {
+                        tmp = g_settings_get_string (manager->priv->settings, keys[i].settings_key);
+                } else {
+                        tmp = g_strdup (keys[i].hard_coded);
+                }
+
+                if (!is_valid_shortcut(tmp))
+                {
+                        g_debug("Not a valid shortcut: '%s'", tmp);
+                        g_free(tmp);
+                        continue;
+                }
+
+                key = g_new0(Key, 1);
+
+                if (!egg_accelerator_parse_virtual(tmp, &key->keysym, &key->keycodes, &key->state))
+                {
+                        g_debug("Unable to parse: '%s'", tmp);
+                        g_free(tmp);
+                        g_free(key);
+                        continue;
+                }
+
+                g_free(tmp);
+
+                keys[i].key = key;
+
+                need_flush = TRUE;
+                grab_key_unsafe(key, TRUE, manager->priv->screens);
+        }
+
+        if (need_flush)
+        {
+                gdk_display_flush (dpy);
+        }
+
+        if (gdk_x11_display_error_trap_pop (dpy))
+        {
+                g_warning("Grab failed for some keys, another application may already have access the them.");
+        }
+
+        mate_settings_profile_end(NULL);
```

Comment:
If you do not make any changes to the code, please do not format it, as you will increase the number of lines to review.